### PR TITLE
handle all file descriptors in parallel

### PR DESCRIPTION
**Pull request by mmsqe**
updates https://github.com/cosmos/cosmos-sdk/issues/15594

speed up ```simd keys list  3.63s user 0.17s system 105% cpu 3.623 total``` to ```simd keys list  0.39s user 0.08s system 62% cpu 0.752 total```

### CODE REVIEW

File: proto/merge.go [hunk 1]
```diff
@@ -48,64 +49,101 @@ func mergedFileDescriptors(debug bool) (*descriptorpb.FileDescriptorSet, error)
...
+	haveFileDescriptor := map[string]bool{}
+	for r := range chFd {
+		if r.Err != nil {
+			return nil, r.Err
 		}
+		fds.File = append(fds.File, r.Fd)
+		haveFileDescriptor[*r.Fd.Name] = true
 	}
+	// Add protoregistry file descriptors to our final file descriptor set.
+	protoregistry.GlobalFiles.RangeFiles(func(fileDescriptor protoreflect.FileDescriptor) bool {
```
Comment (aaronc) [inline proto/merge.go:137]: If we can build an array of these file descriptors first, we could process them in parallel too

**Review by aaronc — COMMENTED** [review]
This looks good. Is there already some test which covers this code to make sure it's working as expected?

File: proto/merge.go (hunk 1 above)
Comment (mmsqe) [reply proto/merge.go:137]: Not sure if I misunderstand, but not much speed gain when I change [RangeFiles](https://github.com/protocolbuffers/protobuf-go/blob/v1.29.1/reflect/protoregistry/registry.go#L350) to
```
func (r *Files) RangeFiles() <-chan protoreflect.FileDescriptor {
	ch := make(chan protoreflect.FileDescriptor)
	if r == nil {
		close(ch)
		return ch
	}
	if r == GlobalFiles {
		globalMutex.RLock()
		defer globalMutex.RUnlock()
	}
	go func() {
		defer close(ch)
		var wg sync.WaitGroup
		for _, files := range r.filesByPath {
			for _, file := range files {
				wg.Add(1)
				go func(file protoreflect.FileDescriptor) {
					defer wg.Done()
					select {
					case ch <- file:
					default:
						return
					}
				}(file)
			}
		}
		wg.Wait()
	}()
	return ch
}
```

**Comment by mark-rushakoff** [comment]
@mmsqe, thanks for opening this PR. I used your approach as inspiration for #61, which is based on more recent commits and includes backfilled tests and benchmarks. I'm closing this PR as #61 supersedes it.

**Pull request closed without merging**